player.go

```diff
@@ -0,0 +1,493 @@
+package spotify
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"strconv"
+)
+
+// PlayerDevice contains information about a device that a user can play music on
+type PlayerDevice struct {
+	// ID of the device. This may be empty.
+	ID string `json:"id"`
+	// IsActive If this device is the currently active device.
+	IsActive bool `json:"is_active"`
+	// IsRestricted Whether controlling this device is restricted. At present if
+	// this is "true" then no Web API commands will be accepted by this device.
+	IsRestricted bool `json:"is_restricted"`
+	// Name The name of the device.
+	Name string `json:"name"`
+	// Type Device type, such as "Computer", "Smartphone" or "Speaker".
+	Type string `json:"type"`
+	// Volume The current volume in percent.
+	Volume int `json:"volume_percent"`
+}
+
+// PlayerDeviceList is a list of devices
+type PlayerDeviceList struct {
+	PlayerDevices []PlayerDevice `json:"devices"`
+}
+
+// PlayerState contains information about the current playback.
+type PlayerState struct {
+	CurrentlyPlaying
+	// Device The device that is currently active
+	Device PlayerDevice `json:"device"`
+	// ShuffleState Shuffle is on or off
+	ShuffleState bool `json:"shuffle_state"`
+	// RepeatState off, track, context
+	RepeatState string `json:"repeat_state"`
+}
+
+// Context is the playback context
+type Context struct {
```

Comment:
Maybe we should call this `PlaybackContext` so that it doesn't seem related to `context.Context`